lib/semian/atomic_enum.rb

```diff
@@ -0,0 +1,37 @@
+module Semian
+  class AtomicEnum < AtomicInteger #:nodoc:
+    undef :increment_by
+    undef :increment
+
+    module AtomicEnumSharedImplementation
+      def initialize(name, permissions, symbol_list)
+        super(name, permissions)
+        initialize_lookup(symbol_list)
+      end
+
+      def value
+        @num_to_sym.fetch(super)
+      end
+
+      def value=(sym)
+        super(@sym_to_num.fetch(sym))
+      end
+
+      private
+
+      def initialize_lookup(symbol_list)
+        # Assume symbol_list[0] is mapped to 0
+        # Cannot just use #object_id since #object_id for symbols is different in every run
+        # For now, implement a C-style enum type backed by integers
+
+        @sym_to_num = {}
+        symbol_list.each.with_index do |sym, idx|
+          @sym_to_num[sym] = idx
+        end
+        @num_to_sym = @sym_to_num.invert
+      end
+    end
+
+    include AtomicEnumSharedImplementation
```

Comment:
I am thinking now that it may be possible to just use `extend Forwardable` and delegate. I usually strayed away from this because in most cases since there is delegation happening, lets say to an instance variable, the instance variable also needs to access content of the wrapper class, so there is back and forth communication, which is not good. However, in this case I think it works okay since the AtomicInteger instance doesn't need to know about the enum abstraction.

It'll just have a huge list of delegators in the end, something like this:
```
def_delegators :@integer, :semid, :shmid, :execute_atomically, :shared?,
                   :destroy, :acquire_memory_object, :bind_init_fn
```